### PR TITLE
refactor (pod log) container valiate

### PR DESCRIPTION
**Pull request by yuzhiquan**
**What type of PR is this?**

 /kind feature
/kind bug

**What this PR does / why we need it**:
- refactor for container valiate
- TestCheckLogLocation testcase should point out pod name

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/registry/core/pod/strategy.go [hunk 1]
```diff
@@ -564,3 +533,29 @@ func PortForwardLocation(
 	}
 	return loc, nodeInfo.Transport, nil
 }
+
+// valiateContainer valiate container is valid for pod, return valid container
+func valiateContainer(container string, pod *api.Pod) (string, error) {
+	if container == "" {
```
Comment (drsantos20) [inline pkg/registry/core/pod/strategy.go]: i would recommend to use `len(container)` == 0 like the line `481`

**Review by drsantos20 — CHANGES_REQUESTED** [review]

File: pkg/registry/core/pod/strategy.go (hunk 1 above)
Comment (yuzhiquan) [reply pkg/registry/core/pod/strategy.go]: > i would recommend to use `len(container)` == 0 like the line `481`

done

File: pkg/registry/core/pod/strategy.go [hunk 2]
```diff
@@ -564,3 +533,29 @@ func PortForwardLocation(
 	}
 	return loc, nodeInfo.Transport, nil
 }
+
+// valiateContainer valiate container is valid for pod, return valid container
```
Comment (wojtek-t) [inline pkg/registry/core/pod/strategy.go]: validateContainer (here and below)

**Review by wojtek-t — COMMENTED** [review]
One nit

/ok-to-test

File: pkg/registry/core/pod/strategy.go (hunk 2 above)
Comment (yuzhiquan) [reply pkg/registry/core/pod/strategy.go]: > validateContainer (here and below)

done,thanks.

**Pull request merged**